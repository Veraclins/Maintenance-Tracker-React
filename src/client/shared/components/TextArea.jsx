
import React from 'react';
import { PropTypes } from 'prop-types';

const TextArea = ({
  name, value, handleChange, inputLabel, required, error, placeholder,
}) => (
  <React.Fragment>
    <div className="field hs-form-field">
      <label className="hs-form-label" htmlFor={name}>
        <span>{inputLabel}{' '}</span>
        {required ? <span className="hs-form-required">*</span> : false}
      </label>
      <div className="input">
        <textarea
          className="hs-input"
          name={name}
          required={required}
          value={value}
          placeholder={placeholder}
          autoComplete="on"
          onChange={handleChange}
        />
      </div>
      {error
        ? (
          <ul className="hs-error-msgs inputs-list">
            <li>
              <label>{error}</label>
            </li>
          </ul>
        ) : false}
    </div>
  </React.Fragment>
);

TextArea.propTypes = {
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string.isRequired,
  required: PropTypes.bool,
  inputLabel: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};

TextArea.defaultProps = {
  required: false,
  error: '',
};

export default TextArea;
