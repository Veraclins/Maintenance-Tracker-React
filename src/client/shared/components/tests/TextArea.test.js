import React from 'react';
import { shallow } from 'enzyme';

import { TextArea } from '../TextArea';

const mockFunction = jest.fn();
describe('Test the Form component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<TextArea
      name="input"
      required
      inputLabel="Text input"
      value=""
      placeholder="enter text"
      autoComplete="on"
      handleChange={mockFunction}
    />);
    expect(wrapper.find('textarea')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
