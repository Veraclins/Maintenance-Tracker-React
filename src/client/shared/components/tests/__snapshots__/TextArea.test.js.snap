// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Form component renders without crashing 1`] = `
<Fragment>
  <div
    className="field hs-form-field"
  >
    <label
      className="hs-form-label"
      htmlFor="input"
    >
      <span>
        Text input
         
      </span>
      <span
        className="hs-form-required"
      >
        *
      </span>
    </label>
    <div
      className="input"
    >
      <textarea
        autoComplete="on"
        className="hs-input"
        name="input"
        onChange={[MockFunction]}
        placeholder="enter text"
        required={true}
        value=""
      />
    </div>
  </div>
</Fragment>
`;
